### PR TITLE
fix(datepicker): catch datepicker click

### LINKED ISSUE #1040: Typeahead click inside modal causes modal to close
### Bug description:
When having a typeahead inside a modal, when I click a selection the  modal closes. 

## Link to minimally-working plunker that reproduces the issue:

http://plnkr.co/edit/y0QWWHxKfn0x5UXjUN6a?p=preview

Plunker is build in the 2 examples from git.

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 2.1.2 (Also tried angular 2.2.0)

ng-bootstrap: alpha 13 (it was not happening in alpha 11 but now it seems to be doing that as well)

Bootstrap: alpha 5

### LINKED ISSUE #1042: When having a date-picker inside a modal, when I select a date the modal closes.
When having a date-picker inside a modal, when I select a date the modal closes.

Example to reproduce issue:
http://plnkr.co/edit/ATyV4MJ2vvoDgCc9W5Wn?p=preview

### PR DESCRIPTION
**Pull request by deeg**
I took the same approach as to fix #1040 with #1041 

Closes #1042

### CODE REVIEW

File: src/datepicker/datepicker-month-view.spec.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import {TestBed, ComponentFixture} from '@angular/core/testing';
 import {createGenericTestComponent, isBrowser} from '../test/common';
+import {By} from '@angular/platform-browser';
```
Comment (deeg) [inline src/datepicker/datepicker-month-view.spec.ts]: Is there a convention on which import block this one goes with? Or are the line breaks here arbitrary?

**Comment by pkozlowski-opensource** [comment]
See https://github.com/ng-bootstrap/ng-bootstrap/pull/1041#issuecomment-260758547 - I'm concerned that this might not be the "right" fix...

**Comment by pkozlowski-opensource** [comment]
@deeg could you please have a look at the alternative approach in #1067? If it looks reasonable to you I will add tests.

**Comment by pkozlowski-opensource** [comment]
I'm going to close this one in favor of #1067 as I believe that we should rather fix modal instead of adjusting each and every directive that could be used in a modal.

**Pull request closed without merging**